pkg/projectfile/projectfield.go

```diff
@@ -99,6 +71,13 @@ func (p *projectField) Save(path string) error {
 		return errs.Wrap(err, "ioutil.ReadFile %s failed", path)
 	}
 
+	// Strip legacy commitID parameter.
+	q := p.url.Query()
+	if q.Has("commitID") {
+		q.Del("commitID")
+		p.url.RawQuery = q.Encode()
+	}
```

Comment:
This feels like the wrong place. I could theoretically make changes to these fields and use these changes, just to have them be changed when I call Save(). Save should only propagate the changes, not itself make changes.